guides/micronaut-oracle-cloud-streaming/chess-game/groovy/src/main/groovy/example/micronaut/chess/GameController.groovy

```diff
@@ -0,0 +1,81 @@
+package example.micronaut.chess
+
+import example.micronaut.chess.dto.GameDTO
+import example.micronaut.chess.dto.GameStateDTO
+import groovy.transform.CompileStatic
+import io.micronaut.http.annotation.Controller
+import io.micronaut.http.annotation.PathVariable
+import io.micronaut.http.annotation.Post
+import io.micronaut.scheduling.TaskExecutors
+import io.micronaut.scheduling.annotation.ExecuteOn
+import io.reactivex.Single
+
+import static io.micronaut.http.MediaType.APPLICATION_FORM_URLENCODED
+import static io.micronaut.http.MediaType.TEXT_PLAIN
+
+/**
+ * Endpoints called by the front end using Ajax.
+ */
+@Controller('/game')
+@ExecuteOn(TaskExecutors.IO) // <1>
+@CompileStatic
+class GameController {
+
+    private final GameReporter gameReporter
+
+    GameController(GameReporter gameReporter) {
+        this.gameReporter = gameReporter
+    }
+
+    /**
+     * Start a new game.
+     *
+     * @param b black's name
+     * @param w white's name
+     * @return the id of the game
+     */
+    @Post(value = '/start', consumes = APPLICATION_FORM_URLENCODED, produces = TEXT_PLAIN) // <2>
```

Comment:
I typically like to use multiple annotations: 

```java
@Post('/start')
@Produces(MediaType.TEXT_PLAIN)
@Consumes(MediaType.APPLICATION_FORM_URLENCODED)
```

That way we can create callouts for those areas.

Reply:
You can create callouts by splitting the members on different lines not sure why more annotations are needed